presto-plugin-toolkit/src/main/java/com/facebook/presto/spi/connector/DelegateConnectorAccessControl.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.connector;
+
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.spi.security.Identity;
+import com.facebook.presto.spi.security.Privilege;
+
+import java.util.Set;
+import java.util.function.Supplier;
+
+import static java.util.Objects.requireNonNull;
+
+public class DelegateConnectorAccessControl
+        implements ConnectorAccessControl
+{
+    private final Supplier<ConnectorAccessControl> delegate;
```

Comment:
https://github.com/prestodb/presto/pull/11762#discussion_r227455397

i feel strongly about it, but i see how it fits your use-case a little better.
Maybe try to avoid introducing those abstractions? Or add private adapter from Refresher concept to Forwarding* concept -- it will be very short ("O(1)") class.